AD419.DataHelper.Web/Models/CfdaNumberImport.cs

```diff
@@ -26,5 +34,35 @@ public CfdaNumberImport(DataRow row)
         [Display(Name = "Program Title")]
         [StringLength(300)]
         public string ProgramTitle { get; set; }
+
+        [Required]
+        [Display(Name = "Agency/Office")]
+        [StringLength(2048)]
+        public string AgencyOffice { get; set; }
+
+        [Display(Name = "Code")]
+        [StringLength(10)]
+        public string Code { get; set; }
+
+        /// <summary>
+        /// Sets the Code based of the Agency/Office for NIFA and NIH.
+        /// </summary>
+        /// <param name="agencyOffice"></param>
+        private string GetCode(string agencyOffice)
+        {
+            if (string.IsNullOrEmpty(agencyOffice))
+                return null;
+
+            if (agencyOffice.StartsWith(USDA) && agencyOffice.Contains(NIFA))

```

Comment:
@taylorkj this compare is case sensitive so I'm not sure if that matters but just so you know.